ui/client/components/modals/GenerateTestDataDialog.js

```diff
@@ -22,14 +25,23 @@ class GenerateTestDataDialog extends React.Component {
   }
 
   render() {
+    const validators = [literalIntegerValueValidator, mandatoryValueValidator]
     return (
       <GenericModalDialog
         init={() => this.setState(this.initState)}
         confirm={this.confirm}
+        okBtnConfig={{disabled: !allValid(validators, [this.state.testSampleSize])}}
         type={Dialogs.types.generateTestData}
       >
         <p>Generate test data</p>
-        <input autoFocus={true} className="add-comment-on-save" value={this.state.testSampleSize} onChange={(e) => { this.setState({testSampleSize: e.target.value}) } }/>
+        <input
+          autoFocus={true}
+          className="add-comment-on-save"
+          value={this.state.testSampleSize}
+          onChange={(e) => { this.setState({testSampleSize: e.target.value}) } }
```

Comment:
This should be defined as method at GenerateTestDataDialog.

Reply:
ok, done